src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperBlock.cs

```diff
@@ -123,6 +124,38 @@ public override IEnumerable<Span> Flatten()
             }
         }
 
+        public override Span LocateOwner(TextChange change)
+        {
+            var oldPosition = change.OldPosition;
+            if (oldPosition < Start.AbsoluteIndex)
+            {
+                // Change occurs prior to the TagHelper.
+                return null;
+            }
+
+            var startTagEndLocation = Start.AbsoluteIndex + SourceStartTag?.Length;
+            if (oldPosition < startTagEndLocation)
```

Comment:
Might be more in line with the previous if check to see if the change is outside bounds here.

```C#
var bodyEndLocation = base.Start.AbsoluteIndex + base.Length;
if (oldPosition > bodyEndLocation)
{
   return null;
}
```